R/slide.R

```diff
@@ -70,51 +70,81 @@
 #' @export
 #' @seealso [`epi_slide_opt`] for optimized slide functions
 #' @examples
+#' library(dplyr)
+#'
 #' # Get the 7-day trailing standard deviation of cases and the 7-day trailing mean of cases
 #' cases_deaths_subset %>%
 #'   epi_slide(
 #'     cases_7sd = sd(cases, na.rm = TRUE),
 #'     cases_7dav = mean(cases, na.rm = TRUE),
 #'     .window_size = 7
 #'   ) %>%
-#'   dplyr::select(geo_value, time_value, cases, cases_7sd, cases_7dav)
+#'   select(geo_value, time_value, cases, cases_7sd, cases_7dav)
+#' # Note that epi_slide_mean could be used to more quickly calculate cases_7dav.
+#'
+#' # In addition to the [`dplyr::mutate`]-like syntax, you can feed in a function or
+#' # formula in a way similar to [`dplyr::group_modify`]:
+#' my_summarizer <- function(window_data) {
+#'   window_data %>%
+#'     summarize(
+#'       cases_7sd = sd(cases, na.rm = TRUE),
+#'       cases_7dav = mean(cases, na.rm = TRUE)
+#'     )
+#' }
+#' cases_deaths_subset %>%
+#'   epi_slide(
+#'     ~ my_summarizer(.x),
+#'     .window_size = 7
+#'   ) %>%
+#'   select(geo_value, time_value, cases, cases_7sd, cases_7dav)
+#'
+#'
+#'
+#'
+#'
+#' #### Advanced: ####
 #'
-#' # The same as above, but unpacking using an unnamed data.frame with a formula
+#' # The tidyverse supports ["packing"][tidyr::pack] multiple columns into a
+#' # single tibble-type column contained within some larger tibble. Like dplyr,
+#' # we normally don't pack output columns together, but will if you provide a
+#' # name for a tibble-type output:
```

Comment:
You get a packed tibble if you do `tibble_col = tibble(.....)` or `~ tibble(.....), new_col_name = "tibble_col"`.  That seems pretty natural/"tidy"(verse) to me, but it's [tripped others up](https://github.com/cmu-delphi/epiprocess/issues/534) (as has the difference between `epi_slide_opt` which is like an `across` vs. the base `epi_slide`).

I don't *think* we want to emphasize this, as we're now trying get people to only use `epi_slide` for transformations, not forecasting, so the typical output should be just simple atomic vectors most of the time.  (And `epix_slide` output column naming behavior should mean that you shouldn't normally have to pack/nest/mass-rename to hide your desired column names from `epix_slide`, because the defaults should be closer to what you want anyway.)

I was debating between two approaches:
- (Current) dump this in an "advanced" section.
- Delete these from examples altogether.

Thoughts?

Reply:
I always appreciate having more examples, so my vote is also for keeping in the "advanced" section. I think the behavior makes sense, but the comment above the example is confusing to me. I think the "how to turn on packing" should be emphasized a little more in the sentence.  Maybe

```suggestion
#' # we normally don't pack output columns together. However, packing behavior can be turned on
#' # by providing a name for a tibble-type output:
```